server/datastore/mysql/mdm_test.go

```diff
@@ -166,6 +168,50 @@ func testMDMCommands(t *testing.T, ds *Datastore) {
 	require.Equal(t, winCmd.TargetLocURI, cmds[1].RequestType)
 	require.Equal(t, "Pending", cmds[1].Status)
 
+	// filter by host Identifier
+	identifiers := map[string]string{
+		windowsH.Hostname:       winCmd.CommandUUID,
+		*windowsH.OsqueryHostID: winCmd.CommandUUID,
+		*windowsH.NodeKey:       winCmd.CommandUUID,
+		windowsH.UUID:           winCmd.CommandUUID,
+		windowsH.HardwareSerial: winCmd.CommandUUID,
+		macH.Hostname:           appleCmdUUID,
+		*macH.OsqueryHostID:     appleCmdUUID,
+		*macH.NodeKey:           appleCmdUUID,
+		macH.UUID:               appleCmdUUID,
+		macH.HardwareSerial:     appleCmdUUID,
+	}
+
+	for identifier, expected := range identifiers {
+		t.Run(identifier, func(t *testing.T) {
+			cmds, err = ds.ListMDMCommands(
+				ctx,
+				fleet.TeamFilter{User: test.UserAdmin},
+				&fleet.MDMCommandListOptions{
+					Filters: fleet.MDMCommandFilters{
+						HostIdentifier: identifier,
+					},
+				},
+			)
+			require.NoError(t, err)
+			require.Len(t, cmds, 1)
+			require.Equal(t, expected, cmds[0].CommandUUID)
+		})
+	}
+
+	// non-existent host identifier
+	cmds, err = ds.ListMDMCommands(
+		ctx,
+		fleet.TeamFilter{User: test.UserAdmin},
+		&fleet.MDMCommandListOptions{
+			Filters: fleet.MDMCommandFilters{
+				HostIdentifier: "non-existent",
```

Comment:
Let's add tests for covering all the combinations of the provided filters:
```
jointStmt, params := getCombinedMDMCommandsQuery(ds, listOpts.Filters.HostIdentifier)
jointStmt += ds.whereFilterHostsByTeams(tmFilter, "h")
jointStmt, params = addRequestTypeFilter(jointStmt, &listOpts.Filters, params)
jointStmt, params = appendListOptionsWithCursorToSQL(jointStmt, params, &listOpts.ListOptions)
```
E.g. test with specifying both host identifier AND request type, etc.